### PR TITLE
Allow passing custom error and label components to LabeledField

### LINKED ISSUE #309: Allow custom label and error label components via props
cc @chawes13

### PR DESCRIPTION
**Pull request by chawes13**
Resolves #309 

This has come up several times recently, especially re: formatting checkbox labels

### CODE REVIEW

**Comment by chawes13** [comment]
@dpikt FYI - flexible on the approach. Figured this would be more of a working review

File: src/forms/labels/labeled-field.js [hunk 1]
```diff
@@ -54,23 +66,28 @@ const defaultProps = {
   hideErrorLabel: false,
 }
 
+const DefaultLabelComponent = InputLabel
+const DefaultErrorComponent = InputError
```
Comment (dpikt) [inline src/forms/labels/labeled-field.js]: I don't know if renaming these as separate vars is adding a lot. But if you want to keep them separate, you should use constant naming: `DEFAULT_LABEL_COMPONENT`

File: src/forms/labels/labeled-field.js [hunk 2]
```diff
@@ -40,6 +42,16 @@ import InputLabel from './input-label'
  *     </LabeledField>
  *   )
  * }
+ * 
+ * // A custom label to pass in as a label component (re-using `<InputLabel />`)
+ * 
+ * const handleClick = () => TermsModal.show()
+ * function CustomLabelComponent (props) {
```
Comment (dpikt) [inline src/forms/labels/labeled-field.js]: You should include an example of passing this to `LabeledPhoneInput`. Also, defining handleClick outside the component it a little unorthodox- perhaps pass it as another prop to `LabeledPhoneInput`?

**Comment by dpikt** [comment]
Might be nice to include some additional `LabeledField` stories along with this PR.

**Comment by chawes13** [comment]
@dpikt I added an example and some stories but would like to get your feedback before merging (haven't added a story before and not sure I accomplished what you were asking for in passing down props)

File: docs.md [hunk 3]
```diff
@@ -1132,14 +1132,25 @@ function LabeledPhoneInput (props) {
...
+ )
 }
 
+const props = {
+  input: { name: 'foo' },
+  meta: { touched: true, invalid: true },
+  labelComponent: CustomLabelComponent,
+}
+
+<LabeledPhoneInput
```
Comment (dpikt) [inline docs.md]: Ah, I was thinking usage with `Field`:

```
<Field
    name="phonenumber"
    component={ LabeledPhoneInput }
    labelComponent={ CustomLabelComponent }
    onClickLabel={ () => 'bar' }
/>
```

**Pull request merged**